pkg/volume/glusterfs/glusterfs.go

```diff
@@ -160,8 +158,33 @@ func (plugin *glusterfsPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volu
 	return plugin.newMounterInternal(spec, ep, pod, plugin.host.GetMounter(plugin.GetPluginName()))
 }
 
+func (plugin *glusterfsPlugin) getEndpointNameAndNamespace(spec *volume.Spec, defaultNamespace string) (string, string, error) {
+	if spec.Volume != nil && spec.Volume.Glusterfs != nil {
+		endpoints := spec.Volume.Glusterfs.EndpointsName
+		if endpoints != "" {
+			return endpoints, defaultNamespace, nil
+		}
+
+	} else if spec.PersistentVolume != nil &&
+		spec.PersistentVolume.Spec.Glusterfs != nil {
+		endpoints := spec.PersistentVolume.Spec.Glusterfs.EndpointsName
+		endpointsNs := defaultNamespace
+
+		if overriddenNs := spec.PersistentVolume.Spec.Glusterfs.EndpointsNamespace; overriddenNs != nil && len(*overriddenNs) > 0 {
```

Comment:
validation ensures this is non-empty if non-nil... if set to "", return an error here.

Reply:
Done.